### PR TITLE
Update JIT format job to pass new `--cross` argument on Linux

### PR DESCRIPTION
**Pull request by BruceForstall**
This argument is passed through to the jit-format tool and is used when invoking build-runtime.sh on Linux when in a cross-build scenario, such as CI builds in Mariner.

Also, improve the jitformat.py script: (1) remove unused code, (2) use jitutil.py, (3) add more error checking, (4) add `--jitutils` argument for easier testing, (5) change from `--coreclr` to `--runtime` argument for specifying repo.

This depends on https://github.com/dotnet/jitutils/pull/380 (and vice-versa). I've tested locally on Windows, Linux (without `--cross`), and Linux/Mariner (with `--cross`).

### CODE REVIEW

**Comment by BruceForstall** [comment]
@kunalspathak https://github.com/orgs/dotnet/teams/jit-contrib PTAL
cc @sbomer

**Review by kunalspathak — APPROVED** [review]
LGTM with few questions.

File: eng/pipelines/coreclr/templates/format-job.yml [hunk 1]
```diff
@@ -50,12 +56,16 @@ jobs:
         version: '6.x'
         includePreviewVersions: true
         installationPath: $(Agent.ToolsDirectory)/dotnet
-    - script: $(PythonScript) $(Build.SourcesDirectory)/src/coreclr/scripts/jitformat.py -c $(Build.SourcesDirectory)/src/coreclr -o $(osGroup) -a $(archType)
+    - script: $(PythonScript) $(Build.SourcesDirectory)/src/coreclr/scripts/jitformat.py -r $(Build.SourcesDirectory) -o $(osGroup) -a $(archType) $(LinuxCrossArg)
+        value: ''
```
[the remark this answers is not in the document]
Comment (BruceForstall) [reply eng/pipelines/coreclr/templates/format-job.yml:60]: oops, that's a weird copy/paste bug.

File: src/coreclr/scripts/jitformat.py [hunk 2]
```diff
@@ -10,234 +10,207 @@
...
             default=None, help='operating system')
-    required.add_argument('-c', '--coreclr', type=str,
-            default=None, help='full path to coreclr')
+    required.add_argument('-r', '--runtime', type=str,
+            default=None, help='full path to runtime repo root')
+
+    optional = parser.add_argument_group('optional arguments')
+    optional.add_argument('--cross', action="store_true",
+            default=None, help='do cross builds on Linux')
+    optional.add_argument('-j', '--jitutils', type=str,
```
[the remark this answers is not in the document]
Comment (BruceForstall) [reply src/coreclr/scripts/jitformat.py:38]: Yes, this makes it easier to test locally

**Pull request merged**